data_steward/cdr_cleaner/cleaning_rules/ensure_date_datetime_consistency.py

```diff
@@ -46,62 +62,98 @@
+END AS {field}
+"""
 
-def get_cols(table_id):
-    """
-    Generates the fields to choose along with case statements to generate datetime
 
-    :param table_id: table for which the fields
-    :return:
+class EnsureDateDatetimeConsistency(BaseCleaningRule):
     """
-    table_fields = field_mapping.get_domain_fields(table_id)
-    col_exprs = []
-    for field in table_fields:
-        if field in TABLE_DATES[table_id]:
-            if field_mapping.is_field_required(table_id, field):
-                col_expr = (
-                    ' CASE'
-                    ' WHEN EXTRACT(DATE FROM {field}) = {date_field}'
-                    ' THEN {field}'
-                    ' ELSE CAST(DATETIME({date_field}, EXTRACT(TIME FROM {field})) AS TIMESTAMP)'
-                    ' END AS {field}').format(
-                        field=field, date_field=TABLE_DATES[table_id][field])
-            else:
-                col_expr = (' CASE'
-                            ' WHEN EXTRACT(DATE FROM {field}) = {date_field}'
-                            ' THEN {field}'
-                            ' ELSE NULL'
-                            ' END AS {field}').format(
-                                field=field,
-                                date_field=TABLE_DATES[table_id][field])
-        else:
-            col_expr = field
-        col_exprs.append(col_expr)
-    cols = ', '.join(col_exprs)
-    return cols
-
-
-def get_fix_incorrect_datetime_to_date_queries(project_id, dataset_id):
+    Ensure no nulls and consistency in the datetime and date fields
     """
-    This function generates a list of query dicts for ensuring the dates and datetimes are consistent
 
-    :param project_id: the project_id in which the query is run
-    :param dataset_id: the dataset_id in which the query is run
-    :return: a list of query dicts for ensuring the dates and datetimes are consistent
-    """
-    queries = []
-    for table in TABLE_DATES:
-        query = dict()
-        query[cdr_consts.QUERY] = FIX_DATETIME_QUERY.format(
-            project_id=project_id,
-            dataset_id=dataset_id,
-            table_id=table,
-            cols=get_cols(table))
-        query[cdr_consts.DESTINATION_TABLE] = table
-        query[cdr_consts.DISPOSITION] = bq_consts.WRITE_TRUNCATE
-        query[cdr_consts.DESTINATION_DATASET] = dataset_id
-        queries.append(query)
-    return queries
+    def __init__(self, project_id, dataset_id, sandbox_dataset_id):
+        """
+        Initialize the class with proper information.
+
+        Set the issue numbers, description and affected datasets. As other tickets may affect this SQL,
+        append them to the list of Jira Issues.
+        DO NOT REMOVE ORIGINAL JIRA ISSUE NUMBERS!
+        """
+        desc = (
+            'Ensure date and datetime consistency where nulls exist in nullable fields'
+            'by setting date in datetime to the date in date field and time as midnight (00,00,00)'
```

Comment:
@dcampbell-vumc you should consider updating this description as well.  This description does accurately reflect what your changes to the rule are doing.  But it doesn't reflect the full extent of what this rule is doing.  This description may be used by reporting when we want to describe how the data was cleaned (i.e. may be seen by people who are not developers).